common/plugins/org.polarsys.capella.common.libraries/src/org/polarsys/capella/common/libraries/manager/LibraryManager.java

```diff
@@ -100,8 +112,14 @@ public void onLibraryProviderChanged(LibraryProviderEvent event) {
   @Override
   public Collection<IModel> getAllModels(TransactionalEditingDomain domain) {
     HashMap<IModelIdentifier, IModel> models = new HashMap<IModelIdentifier, IModel>();
+    List<AirdResource> airdResources = domain.getResourceSet().getResources().stream().filter(AirdResource.class::isInstance).map(AirdResource.class::cast).collect(Collectors.toList());
+    Iterator<AirdResource> airdResourcesIterator = airdResources.iterator();
+    Session session = null;
+    while (session == null && airdResourcesIterator.hasNext()) {
+        session = SessionManager.INSTANCE.getSession(airdResourcesIterator.next().getURI(), new NullProgressMonitor());
+    }
 
-    for (IModelIdentifier identifier : getAvailableModels()) {
+    for (IModelIdentifier identifier : getAvailableModels(session)) {
```

Comment:
domain

Reply:
I used session instead of domain here because in com.thalesgroup.mde.melody.collab.ui.provider.CDOLibraryProvider.getAvailableModels(Session) the session is used twice but there is no api for the domain. I could find the session from the domain though. Did you have a specific reason to want this new API to be with domain instead of session?

Reply:
I have my answer from the next comment.
Fixed